File: frontend/lib/domain/entities/carroceria.dart
class Carroceria {
  final int? id;
  final String nombre;
  final String? descripcion;
  final int claseId;

  Carroceria(
      {this.id,
      required this.nombre,
      required this.descripcion,
      required this.claseId});

  factory Carroceria.fromJson(Map<String, dynamic> json) {
    return Carroceria(
        id: json['id'],
        nombre: json['nombre'],
        descripcion: json['descripcion'],
        claseId: json['claseId']);
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'descripcion': descripcion,
      'claseId': claseId
    };
  }
}

File: backend/Mover.Entities/POCOEntities/Vehiculos/Carroceria.cs
namespace Mover.Entities.POCOEntities.Vehiculos
{
    public class Carroceria
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = string.Empty;
        public string? Descripcion { get; set; }

        public int? ClaseId { get; set; }
        public Clase? Clase { get; set; }
    }
}
